tests/datasets/test_geo.py

```diff
@@ -178,6 +178,26 @@ def sentinel(self, request: SubRequest) -> Sentinel2:
         cache = request.param[1]
         return Sentinel2(root, bands=bands, transforms=transforms, cache=cache)
 
+    def test_init_with_list_of_files(self) -> None:
```

Comment:
Added this quick and dirty test for proof of concept. Open to suggestions on how this should be tested.